src/main/java/org/opensearch/ad/rest/handler/ModelValidationActionHandler.java

```diff
@@ -0,0 +1,755 @@
+        client.search(searchRequest, ActionListener.wrap(response -> processTopEntityResults(response, latestTime), listener::onFailure));
+    }
+
+    private void processTopEntityResults(SearchResponse response, long latestTime) {
+        Histogram aggregate = checkBucketResultErrors(response);
+        if (aggregate == null) {
+            return;
+        }
+        double fullBucketRate = processBucketAggregationResults(aggregate);
+        if (fullBucketRate < CONFIG_BUCKET_MINIMUM_SUCCESS_RATE) {
+            listener
+                .onFailure(
+                    new ADValidationException(
+                        CommonErrorMessages.CATEGORY_FIELD_TOO_SPARSE,
+                        DetectorValidationIssueType.CATEGORY,
+                        ValidationAspect.MODEL
+                    )
+                );
+        } else {
+            try {
+                checkFeatureQuery(latestTime);
+            } catch (Exception ex) {
+                logger.error(ex);
+                listener.onFailure(ex);
+            }
+        }
+    }
+
+    private void processFeatureQuery(SearchResponse response, long latestTime) {
+        Histogram aggregate = checkBucketResultErrors(response);
+        if (aggregate == null) {
+            return;
+        }
+
+        double fullBucketRate = processBucketAggregationResults(aggregate);
+        if (fullBucketRate < CONFIG_BUCKET_MINIMUM_SUCCESS_RATE) {
+            listener
+                .onFailure(
+                    new ADValidationException(
+                        CommonErrorMessages.FEATURE_QUERY_TOO_SPARSE,
+                        DetectorValidationIssueType.FEATURE_ATTRIBUTES,
+                        ValidationAspect.MODEL
+                    )
+                );
+            return;
+        }
+        windowDelayRecommendation(latestTime);
+    }
+
+    private void checkFeatureQuery(long latestTime) throws IOException {
+        List<String> featureFields = ParseUtils.getFeatureFieldNames(anomalyDetector, xContentRegistry);
+        AggregationBuilder aggregation = getBucketAggregation(
+            latestTime,
+            (IntervalTimeConfiguration) anomalyDetector.getDetectionInterval()
+        );
+        BoolQueryBuilder query = QueryBuilders.boolQuery().filter(anomalyDetector.getFilterQuery());
+        for (String featureField : featureFields) {
+            query.filter(QueryBuilders.existsQuery(featureField));
+        }
+        SearchSourceBuilder searchSourceBuilder = getSearchSourceBuilder(query, aggregation);
+        SearchRequest searchRequest = new SearchRequest(anomalyDetector.getIndices().toArray(new String[0])).source(searchSourceBuilder);
+        client.search(searchRequest, ActionListener.wrap(response -> processFeatureQuery(response, latestTime), listener::onFailure));
+    }
+
+    private void sendWindowDelayRec(long latestTime) {
+        long minutesSinceLastStamp = TimeUnit.MILLISECONDS.toMinutes(Instant.now().toEpochMilli() - latestTime);
```

Comment:
How does rounding work in this case? For example, if the difference is 1400 ms -> 1.4 minutes, will it return 1 or 2?. In this case, 1 minute isn't enough delay, and 2 minutes may be a better suggestion to guarantee incoming data.